backEnd/app/get_search_result.py

```diff
@@ -929,18 +929,19 @@ def get_annId_song_list(
 
     print("-------------------------")
     print("Date: ", str(datetime.now().strftime("%d/%m/%Y %H:%M:%S")))
-    print(f"annId_filter: {annId}")
+    print(f"annId_filter: {annIds}")
     print(f"ignore_dups: {ignore_duplicate}", end=" | ")
     print(f"types: {authorized_types}", end=" | ")
     print(f"broadcasts: {authorized_broadcasts}", end=" | ")
     print(f"song_categories: {authorized_song_categories}")
 
-    if not str(annId).isdigit():
-        return []
+    if len(annIds) == 1:
+        if not str(annIds[0]).isdigit():
+            return []
```

Comment:
If you're allowing multiple IDs you'd also want to validate the whole array of IDs no matter the length, not just if the the amount of IDs is exactly 1. 
Like this, issues would arise if no ID is passed and if 2+ ids are passed. 

Consider this instead? 
```suggestion
    if len(annIds) == 0:
        return []

    if not all(str(annId).isdigit() for annId in annIds):
        return []
```

Reply:
Thank you for the clarification. I just wasn't sure what usage the check was for, so I defaulted to only have it done in the case where the previous code would have used it (if that makes any sense).
I should probably have thought of executing it on every array element instead, precisely since I didn't know what it was for... my bad.